### PR TITLE
Easy CSV Visualization

### PR DESCRIPTION
**Pull request by fa9r**
I added a `zenml.types.CSVString` special string class that can be returned from steps to automatically visualize the corresponding string as CSV in the dashboard.

### Usage Example:

```python
from zenml.types import CSVString

@step
def my_step() -> CSVString:
    some_csv = "a,b,c\n1,2,3"
    return CSVString(some_csv)
```

### Resulting Visualization in Dashboard:

Screenshot 2023-05-23 at 22 50 27

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/materializers/special_string_materializer.py [hunk 1]
```diff
@@ -20,24 +20,25 @@
...
 from zenml.logger import get_logger
 from zenml.materializers.base_materializer import BaseMaterializer
-from zenml.types import HTMLString, MarkdownString
+from zenml.types import CSVString, HTMLString, MarkdownString
 
 logger = get_logger(__name__)
 
 
-HTML_OR_MARKDOWN = Union[HTMLString, MarkdownString]
+SPECIAL_STRINGS = Union[CSVString, HTMLString, MarkdownString]
```
Comment (strickvl) [inline src/zenml/materializers/special_string_materializer.py:28]: How about 'STRUCTURED_STRINGS' instead? Somehow SPECIAL could be lots of things, like those containing special characters etc.

**Review by strickvl — COMMENTED** [review]
Broadly fine with this, though wondering a bit about the name. Also will probably want a Docs Update alongside, otherwise people won't use these materialisers, no?

**Review by htahir1 — COMMENTED** [review]
I dont have any idea how this works so im just maybe leave a question: Is this bakcwards- compatible? :-D

**Comment by fa9r** [comment]
@htahir1 for existing visualizations yes. However, loading artifacts saved by the materializer is no longer possible since the code reference to the materializer has changed by renaming it. That will always happen if you rename/move materializers though.

**Review by strickvl — APPROVED** [review]

**Review by AlexejPenner — APPROVED** [review]

**Pull request merged**